tests/features/bootstrap/FeatureContext.php

```diff
@@ -5,85 +5,144 @@
     Behat\Behat\Context\BehatContext,
     Behat\Behat\Exception\PendingException;
 use Behat\Gherkin\Node\PyStringNode,
-    Behat\Gherkin\Node\TableNode;
-
-use VCR\VCR;
+    Behat\Gherkin\Node\TableNode,
+    Behat\Gherkin\Node\ScenarioNode;
 
 /**
  * Features context.
  */
-class FeatureContext extends BehatContext
-{
+class FeatureContext extends BehatContext {
     public $cliroot = '';
-    private $testroot = 'tests';
+    private $cassette_name;
+    private $connection_info = [];
+    private $output;
+
     /**
-     * Initializes context.
-     * Every scenario gets it's own context object.
-     *
-     * @param array $parameters context parameters (set them up through behat.yml)
+    * Initializes context. Sets directories for navigation.
+    * 
+    * @param $parameters [array] context parameters (set them up through behat.yml)
+    */
+    public function __construct(array $parameters) {
+      $this->cliroot = dirname(dirname(__DIR__)) . '/..';
+      $this->connection_info = $parameters;
+    }
+
+    /**
+    * @BeforeScenario
+    * Runs before each scenario
+    * 
+    * @param $event [ScenarioEvent]
+    */
+    public function before($event) {
+      $this->setCassetteName($event);
+    }
+
+    /**
+     * @When /^I am authenticating$/
      */
-    public function __construct(array $parameters)
-    {
-        $this->cliroot = dirname(dirname(__DIR__));
-        
-        // Initialize your context here
+    public function iAmAuthenticating() {
+      if(!isset($this->connection_info['username']) || !isset($this->connection_info['password'])) 
+        throw new Exception("Check your configuration file to ensure proper configuration.");
+      $this->iRun('terminus auth login ' . $this->connection_info['username'] . ' --password=' . $this->connection_info['password']);
     }
 
-    /** @Given /^I am in directory "([^"]*)"$/ */
-    public function iAmInDirectory( $dir )
-    {
-      if( !file_exists('bin/terminus') ) {
-        //must be in the wrong directory so chdir
-        chdir($this->cliroot);
-      }
+    /**
+    * @Given /^I am in directory "([^"]*)"$/
+    * Changes the directory to given subdir of Terminus root directory
+    * 
+    * @param $dir [string]
+    */
+    public function iAmInDirectory($dir) {
+      chdir($this->cliroot . $dir);
       return true;
+    }
+
+    /**
+    * @Then /^I enter "([^"]*)"$/ 
+    * 
+    * @param $string [string]
+    */
+    public function iEnter($string) {
+      $fh = fopen("php://stdin", 'w');
+      fwrite($fh, "$string\n");
+    }
 
+    /**
+    * @When /^I run "([^"]*)"$/
+    * Runs command and saves output
+    * 
+    * @param $command [string]
+    */
+    public function iRun($command) {
+      $terminus_cmd = sprintf('bin/terminus', $this->cliroot);
+      $command = 'VCR_CASSETTE=' . $this->cassette_name 
+        . ' ' . str_replace("terminus", $terminus_cmd, $command) . ' --nocache';
```

Comment:
Can you please remove the `--nocache` flag? We should be running the actual terminus command because the behavior can be drastically different (and broken) with or without that flag.